extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -158,7 +170,13 @@ export class Bind {
     user().assert(this.enabled_, `Experiment "${TAG}" is disabled.`);
 
     // TODO(choumx): What if `state` contains references to globals?
-    Object.assign(this.scope_, state);
+    if (Object.keys(this.scope_).length > 0) {
```

Comment:
I'm not sure that's necessary. In fact I'm not even sure having this if-statement is worth the extra LOC. If the scope and the state being merged in don't both have the same key mapped to objects then there won't be any recursive calls. This applies for when the scope is empty, and for initial merging for amp-state elements since no two amp-state elements can have the same id